### DISCUSSION TITLE
Poor performance with video links?

### DISCUSSION THREAD

**Discussion by nonoumasy** [vercel/next.js#24902] [Help] [answered]
I'm getting poor performance on my Nextjs site:
[https://traveltips.vercel.app/](https://traveltips.vercel.app/)
...due to FCP. 

1. I don't know which domains to add:  Since this is a user-generated site, I can't take advantage of the Image component since I can't determine which domains the images links will come from. 
2. Video links: Some of the content are Youtube videos which are added using embed links. I believe this is what is causing the hit in performance in FCP. Are there any strategies/solution to improving the performance.

Any suggestions?

![Screen Shot 2021-05-07 at 4 13 53 PM](https://user-images.githubusercontent.com/6312676/117517056-aa566d00-af4f-11eb-8c02-353ec54265a3.png)

**Comment by bobarros** [comment, marked as answer]
Don't load the videos on page load. You will show an image and something that seems like a video. 
There's a lot on the internet about it. First hit --> https://css-tricks.com/lazy-load-embedded-youtube-videos/

Only when users click the button that you have an image.
It's simple to do it with Youtube in the desktop version because autoplay will result in no difference for the user.
Mobile can be more tricky.

If you need further help with the code, let me know.

**Reply by nonoumasy** [reply]
Thank you! I didn't know that trick with iframe. I think the analytics improved already. :)
Really thanks a lot. I've been stuck on this for over a week.

**Reply by bobarros** [reply]
Hahaha
A week it's a long time. I'm glad that I was able to help
